Ventivader/Ventivader/Utilities/ErrorScreenProvider.swift

```diff
@@ -0,0 +1,146 @@
+        populateErrorView(title: title,
+                          body: body,
+                          buttonTitle: buttonTitle,
+                          imageView: imageView)
+        
+        view.addSubview(containerView)
+    }
+    
+    func dismissCurrentErorView() {
+        DispatchQueue.main.async { [weak self] in
+            self?.containerView.removeFromSuperview()
+        }
+    }
+    
+    @objc private func actionButtonTap(){
+        actionButtonTapClosure?()
+    }
+    
+    private func populateErrorView(title:String,
+                                 body: String,
+                                 buttonTitle: String?,
+                                 imageView: UIImageView) {
+        
+        let titleLabel  = UILabel()
+        titleLabel.updateFont(name: VentivaderFonts.titleFont, size: 25)
+        titleLabel.adjustsFontSizeToFitWidth = true
+        titleLabel.minimumScaleFactor = 0.5
+        titleLabel.text = title
+        titleLabel.textAlignment = .center
+        
+        let bodyLabel  = UILabel()
+        bodyLabel.updateFont(name: VentivaderFonts.bodyFont)
+        bodyLabel.text = body
+        bodyLabel.textAlignment = .center
+        bodyLabel.numberOfLines = 0
+        
+        let backgroundView = UIView()
+        backgroundView.backgroundColor = ColorPallete.errorBackground
+        
+        let waveImageView = UIImageView()        
+        waveImageView.image = UIImage(named: "errorWave")
+        
+        let actionButton = UIButton()
+        actionButton.titleLabel?.updateFont(name: VentivaderFonts.bodyFont)
+        actionButton.backgroundColor = ColorPallete.errorBackground
+        actionButton.contentEdgeInsets = UIEdgeInsets(top: 5.0, left: 10.0, bottom: 5.0, right: 10.0)
+        actionButton.layer.cornerRadius = 4.0
+        
+        if let buttonTitle = buttonTitle {
+            actionButton.setTitle(buttonTitle, for: .normal)
+        } else {
+            actionButton.isHidden = true
+        }
+        
+        actionButton.addTarget(self, action: #selector(actionButtonTap), for: .touchUpInside)
+        
+        containerView.addSubview(backgroundView)
+        containerView.addSubview(waveImageView)
+        containerView.addSubview(imageView)
+        containerView.addSubview(titleLabel)
+        containerView.addSubview(bodyLabel)
+        containerView.addSubview(actionButton)
+        
+        backgroundView.translatesAutoresizingMaskIntoConstraints = false
+        waveImageView.translatesAutoresizingMaskIntoConstraints = false
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+        titleLabel.translatesAutoresizingMaskIntoConstraints = false
+        bodyLabel.translatesAutoresizingMaskIntoConstraints = false
+        actionButton.translatesAutoresizingMaskIntoConstraints = false
+        
+        let safeContainerArea = containerView.safeAreaLayoutGuide
+        
+        // Background Image
+        NSLayoutConstraint.activate([
```

Comment:
do you prefer to add autolayout constraints in code? either way is fine. Nice :)

Reply:
I think this is how we do the error screens, in other apps ;) (wink).
So, i just keep that approach.

Reply:
Can we simplify this by adding the Error screen into the storyboard or an xib?

Reply:
Ok, i moved this to an xib.